From ecefaead96fce52d499dccdb02191b3e93352231 Mon Sep 17 00:00:00 2001
From: Michael Henning <drawoc@darkrefraction.com>
Date: Sat, 30 Nov 2013 21:31:57 +0000
Subject: Bug 719593 - On clang, brush outline is not displayed

Don't access path_data->len after path_data has been freed.

(cherry picked from commit 95becc7615c7e9cf2f6135c8d5b0fe1cca86648f)
---
diff --git a/app/core/gimpbezierdesc.c b/app/core/gimpbezierdesc.c
index 03ea2f0..f2eacb2 100644
--- app/core/gimpbezierdesc.c
+++ app/core/gimpbezierdesc.c
@@ -111,6 +111,7 @@ gimp_bezier_desc_new_from_bound_segs (BoundSeg *bound_segs,
   gint         n_points;
   gint         seg;
   gint         i;
+  guint        path_data_len;
 
   g_return_val_if_fail (bound_segs != NULL, NULL);
   g_return_val_if_fail (n_bound_segs > 0, NULL);
@@ -159,8 +160,10 @@ gimp_bezier_desc_new_from_bound_segs (BoundSeg *bound_segs,
 
   g_free (points);
 
+  path_data_len = path_data->len;
+
   return gimp_bezier_desc_new ((cairo_path_data_t *) g_array_free (path_data, FALSE),
-                               path_data->len);
+                               path_data_len);
 }
 
 void
--
cgit v0.9.2
